x-pack/plugins/fleet/server/services/api_keys/index.ts

```diff
@@ -16,20 +18,23 @@ export * from './enrollment_api_key';
 export async function generateOutputApiKey(
   soClient: SavedObjectsClientContract,
   outputId: string,
-  agentId: string
+  agentId: string,
+  permissions: { [role: string]: { indices: FullAgentPolicyPermission[] } }
 ): Promise<{ key: string; id: string }> {
   const name = `${agentId}:${outputId}`;
-  const key = await createAPIKey(soClient, name, {
-    'fleet-output': {
+
+  const APIKeyRequest = Object.entries(permissions).reduce<
+    Record<string, { cluster: string[]; indices: FullAgentPolicyPermission[] }>
+  >((request, [role, indices]) => {
+    request[role] = {
       cluster: ['monitor'],
```

Comment:
For my information why the cluster part is not in the permissions object?

Reply:
I tried to only include in the schema what affects the permissions that the integrations might require. I think integrations will need specific permissions in indices and not so much at the cluster level, so I decided to leave it out.

Of course if this is wrong I can include it

Reply:
Alright, I had a quick call with @ruflin and we agreed we should include it in the policy.